tornadox/solvers/ek1.py

```diff
@@ -189,3 +189,232 @@ def extract_qoi_fn(t, state):
         return t, state.u, state.stats
 
     return init_fn, perform_step_fn, extract_qoi_fn
+
+
+def ek1_saveat(
+    *,
+    ode_dimension,
+    num_derivatives=5,
+):
+    """The EK1, but only for computing the checkpoints/saveat-values.
+
+    Uses adaptive steps and PI control.
+    Uses time-varying, scalar-valued diffusion.
+    Uses Taylor-mode initialisation for num_derivatives > 5,
+    and forward-mode initialisation otherwise.
+    """
+
+    # Create the identity matrix required for Kronecker-type things below.
+    eye_d = jnp.eye(ode_dimension)
+    a_1d, q_sqrtm_1d = ibm.system_matrices_1d(num_derivatives=num_derivatives)
+    a, q_sqrtm = jnp.kron(eye_d, a_1d), jnp.kron(eye_d, q_sqrtm_1d)
+
+    EK1State = namedtuple(
+        "EK1State", ("u", "dt_proposed", "backward_model", "error_norm", "stats")
+    )
+
+    # Initialisation with autodiff
+    if num_derivatives <= 5:
+        autodiff_fun = autodiff_first_order.forwardmode_jvp
+    else:
+        autodiff_fun = autodiff_first_order.taylormode
+
+    # At each checkpoint, the backward model needs to be 'reset'.
+    # This means that it is set to an identity, noise-free transition.
+    # We just assemble this once and keep referencing it below.
```

Comment:
Does this actually provide a benefit in jax? Since everything is static anyways I'd have assumed that it doesn't change much, and in that case just calling the identity_transition directly might make it more readable.

EDIT: Actually, maybe just renaming helps already. What do you think of making "identity" more prominent? Maybe it doesn't even need to be called "bw_model", since it is only used when assigned to keyword values `backward_model`?

Reply:
Cool, thanks!